src/indicators/bollinger_bands.rs

```diff
@@ -0,0 +1,230 @@
+/// The Bollinger Badns are represented by Average EMA and standard deviaton that is moved 'k' times away in both directions from calculated average value.
+///
+/// # Formula
+///
+/// Bollinger Bands are calculated based on EMA combined with Standard Deviaiation(SD).
+///
+/// See EMA doumentation.
+///
+/// ![SD formula](https://wikimedia.org/api/rest_v1/media/math/render/svg/8715d659116bec91d48762b9e1f3d9aed36fc028)
+///
+/// Where:
+///
+/// * _SD<sub>s</sub>_ - is value of standard deviation for N given probes.
+/// * _SD<sub>ٍx</sub>_  - is the mean value of observation.
+/// * _SD<sub>N</sub>_ - is number of probes in observation.
+/// * _SD<sud>xi</sub>_ - is i-th observed value from N elements observation.
+///
+/// and then BB is composed as:
+///
+///  * _BB<sub>Middle Band</sub>_ - Exponential Moving Average (EMA).
+///  * _BB<sub>Upper Band</sub>_ = N * (EMA + SD of observation * multipler (usually 2.0))
+///  * _BB<sub>Lower Band</sub>_ = N * (EMA - SD of observation * multipler (usually 2.0))
+///
+/// # Example
+///
+///```
+/// use ta::indicators::{BollingerBands, BollingerBandsOutput};
+/// use ta::Next;
+///
+/// let mut bb = BollingerBands::new(3, 2.0_f64).unwrap();
+///
+/// let out_0 = bb.next(2.0);
+///
+/// let out_1 = bb.next(5.0);
+///
+/// assert_eq!(out_0.average, 2.0);
+/// assert_eq!(out_0.upper, 2.0);
+/// assert_eq!(out_0.lower, 2.0);
+///
+/// assert_eq!(out_1.average, 3.5);
+/// assert_eq!(out_1.upper, 7.742640687119285);
+/// assert_eq!(out_1.lower, -0.7426406871192848);
+/// ```
+///
+/// # Links
+///
+/// ![Bollinger Bands, Wikipedia](https://en.wikipedia.org/wiki/Bollinger_Bands)
+///
+///
+
+#[derive(Debug, Clone)]
+pub struct BollingerBandsOutput {
+    pub average: f64,
+    pub upper: f64,
+    pub lower: f64,
+}
+
+impl BollingerBandsOutput {
+    pub fn new(average: f64, upper: f64, lower: f64) -> Self {
+        Self {average, upper, lower}
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct BollingerBands {
+    length: usize,
+    distance_multiplier: f64,
+    values: Vec<f64>,
+    average: ExponentialMovingAverage,
+}
+
+impl BollingerBands {
+    pub fn new(length: usize, distance_multiplier: f64) -> Result<Self> {
+        if length < 2 {
+            return Err(Error::from_kind(ErrorKind::InvalidParameter));
+        }
+        Ok(Self {
+            length,
+            distance_multiplier,
+            values: Vec::new(),
+            average: ExponentialMovingAverage::new(length as u32).unwrap(),
+        })
+    }
+
+    pub fn length(&self) -> usize {
+        self.length
+    }
+
+    pub fn distance_multiplier(&self) -> f64 {
+        self.distance_multiplier
+    }
+}
+
+impl Next<f64> for BollingerBands {
+    type Output = BollingerBandsOutput;
+
+    fn next(&mut self, input: f64) -> Self::Output {
+        self.values.push(input);
+        let mean = self.average.next(input);
```

Comment:
Ah.. Ok, I think at this point we did it wrong.
From statistics point of view, mean is an average. Considering this, using SMA would be more correct.